src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/AbstractDocumentSpanEntry.cs

```diff
@@ -86,6 +104,25 @@ public override bool TryCreateColumnContent(string columnName, out FrameworkElem
             }
 
             protected abstract IList<Inline> CreateLineTextInlines();
+
+            public static async Task<DocumentSpan> MapAndGetFirstAsync(DocumentSpan documentSpan, CancellationToken cancellationToken)
+            {
+                var service = documentSpan.Document.State.DocumentServices?.GetService<ISpanMapperService>();
```

Comment:
We should make an internal method on Document so we're not grovelling throguh Document.State.DocumentServices like this.

Reply:
(also consider maybe just making that a place that does the null check for DocumentServices)

Reply:
sure. i can make it exposed on Document. I might put DefaultImplementation of DocumentServices which return null for all services so that one doesn't need to check null.